src/toSchema.ts

```diff
@@ -0,0 +1,119 @@
+import type { Runtype } from './runtype'
+import { toSchema as anyToSchema } from './any'
+import { toSchema as arrayToSchema } from './array'
+import { toSchema as booleanToSchema } from './boolean'
+import { toSchema as customToSchema } from './custom'
+import { toSchema as dictionaryToSchema } from './dictionary'
+import { toSchema as enumToSchema } from './enum'
+import { toSchema as guardedByToSchema } from './guardedBy'
+import { toSchema as ignoreToSchema } from './ignore'
+import { toSchema as integerToSchema } from './integer'
+import { toSchema as intersectionToSchema } from './intersection'
+import { toSchema as jsonToSchema } from './json'
+import { toSchema as literalToSchema } from './literal'
+import { toSchema as nullToSchema } from './null'
+import { toSchema as nullOrToSchema } from './nullOr'
+import { toSchema as numberToSchema } from './number'
+import { toSchema as objectToSchema } from './object'
+import { toSchema as optionalToSchema } from './optional'
+import { toSchema as recordToSchema } from './record'
+import { toSchema as stringToSchema } from './string'
+import { toSchema as stringAsIntegerToSchema } from './stringAsInteger'
+import { toSchema as stringLiteralUnionToSchema } from './stringLiteralUnion'
+import { toSchema as tupleToSchema } from './tuple'
+import { toSchema as unionToSchema } from './union'
+import { toSchema as undefinedToSchema } from './undefined'
+import { toSchema as undefinedOrToSchema } from './undefinedOr'
+import { toSchema as unknownToSchema } from './unknown'
+import type { Meta } from './runtypeMeta'
+
+function assertNever(x: never): never {
+  throw new Error(`unexpected object ${x}`)
+}
+
+export function toSchema(runtype: Runtype<any>): string {
```

Comment:
Ok now I see it, this is the actual method implementing the schema generation.

It seems to me that this method *only* needs the `meta`data for each runtype to generate the schema?

Of course it will become quite huge but this way you can supply different implementations quite easily.

Having a single `toSchema` in each runtype seems redundant to me. Also, we would need a `toTypescriptSchema`, `toJsonSchema` and so on for each implementation.